### DISCUSSION TITLE
Groq Client

### DISCUSSION THREAD

**Discussion by Gal-Gilor** [microsoft/autogen#2757] [Ideas]
Hey, 

Despite the similarities with the OpenAI API, I struggled using Autogen with Groq, so I wrote a dedicated client (like Gemini).
I don't have experience contributing. What would be the best path forward? Should I raise an issue? Push a PR?

**Comment by marklysze** [comment]
Hi @Gal-Gilor, there are currently a few client class PRs for Mistral, Anthropic, and Together.AI. Would be great to have Groq, too. You can check out those PRs as a reference for what needs to be created / edited. They are based on the Gemini one.

PRs #2931, #2892, #2919

If you do want to create one then can you also make a note of it in the roadmap, Issue #2946.

Thanks!
